pkg/peertls/tlsopts/tls.go

```diff
@@ -37,25 +45,46 @@ func (opts *Options) ServerOption() grpc.ServerOption {
 }
 
 // DialOption returns a grpc `DialOption` for making outgoing connections
-// to the node with this peer identity
-// id is an optional id of the node we are dialing
-func (opts *Options) DialOption(id storj.NodeID) grpc.DialOption {
+// to the node with this peer identity.
+// id is an optional id of the node we are dialing.
+func (opts *Options) DialOption(ctx context.Context, id storj.NodeID) grpc.DialOption {
+	return grpc.WithTransportCredentials(opts.TransportCredentials(id))
+}
+
+// TransportCredentials returns a grpc `credentials.TransportCredentials`
+// implementation for use within peertls.
+func (opts *Options) TransportCredentials(id storj.NodeID) credentials.TransportCredentials {
+	return &TransportCredentials{
+		TransportCredentials: credentials.NewTLS(opts.TLSConfig(id)),
+	}
+}
+
+// TLSConfig returns a TSLConfig for use in handshaking with a peer.
+func (opts *Options) TLSConfig(id storj.NodeID) *tls.Config {
 	pcvFuncs := append(
 		[]peertls.PeerCertVerificationFunc{
 			peertls.VerifyPeerCertChains,
 			verifyIdentity(id),
 		},
 		opts.PCVFuncs...,
 	)
-	tlsConfig := &tls.Config{
+	return &tls.Config{
 		Certificates:       []tls.Certificate{*opts.Cert},
 		InsecureSkipVerify: true,
 		VerifyPeerCertificate: peertls.VerifyPeerFunc(
 			pcvFuncs...,
 		),
 	}
+}
 
-	return grpc.WithTransportCredentials(credentials.NewTLS(tlsConfig))
+// ClientHandshake returns a non-temporary error if an error is returned from
+// the underlying `ClientHandshake` call.
+func (tc *TransportCredentials) ClientHandshake(ctx context.Context, authority string, conn net.Conn) (net.Conn, credentials.AuthInfo, error) {
+	tlsConn, authInfo, err := tc.TransportCredentials.ClientHandshake(ctx, authority, conn)
+	if err != nil {
```

Comment:
Maybe we should make sure this error is what we think it is to allow for proper functionality on the event of a temporary error (whatever that might be).